src/Storages/MergeTree/MergeTreeSettings.h

```diff
@@ -144,6 +144,7 @@ struct Settings;
     M(Bool, allow_remote_fs_zero_copy_replication, false, "Don't use this setting in production, because it is not ready.", 0) \
     M(String, remote_fs_zero_copy_zookeeper_path, "/clickhouse/zero_copy", "ZooKeeper path for Zero-copy table-independet info.", 0) \
     M(Bool, remote_fs_zero_copy_path_compatible_mode, false, "Run zero-copy in compatible mode during conversion process.", 0) \
+    M(Bool, force_select_final, false, "Query with the FINAL modifier by default", 0) \
```

Comment:
I think if we will have that setting enabled by default globally of for some tables, it would be nice to have also session-level setting to see the raw data (i.e. disable FINAL temporary in the query, for example for debugging - something like 
`ignore_force_select_final` in Settings.h). 

So if table `t` was created with `force_select_final`  

```sql
SELECT * FROM t; -- is actually SELECT * FROM t FINAL; because of force_select_final
--but
SET ignore_force_select_final=1;
SELECT * FROM t; -- now we can look of raw data again w/o finalizing it. 
```

P.S. MergeTreeSetting can not be toggled as easy.

Reply:
@arthurpassos i see you've added the setting, can you please add also a test?

Smth like 
```
CREATE TABLE tbl (id UInt64, val String) engine=ReplacingMergeTree() ORDER BY id SETTINGS force_select_final=1;
SYSTEM STOP MERGES tbl;
INSERT INTO tbl SELECT number as id, 'foo' AS val FROM numbers(100000);
INSERT INTO tbl SELECT number as id, 'bar' AS val FROM numbers(100000);

SELECT * FROM tbl WHERE id = 10000; -- single row expected (bar), because the force_select_final is in action
SELECT * FROM tbl final WHERE id = 10000;  -- single row expected (bar), because the force_select_final is in action and FINAL is there.

SELECT * FROM tbl WHERE id = 10000 SETTINGS ignore_force_select_final=1; -- now we see 2 'real' rows 
SELECT * FROM tbl FINAL WHERE id = 10000 SETTINGS ignore_force_select_final=1; -- now we see single row again.

SYSTEM START MERGES tbl;
OPTIMIZE TABLE tbl FINAL;

SELECT * FROM tbl WHERE id = 10000 SETTINGS ignore_force_select_final=1; -- now we see single row
DROP TABLE tbl;
```